src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,926 @@
+            bsoncxx::builder::stream::document document{};
+            doc->view(document);
+            _writeOps.push_back(mongocxx::model::insert_one{document.extract()});
+        } else if (writeCommand == "deleteOne") {
+            auto filter = writeOp["Filter"];
+            if (!filter) {
+                throw InvalidConfigurationException("'deleteOne' expects a 'Filter' field.");
+            }
+            auto doc = value_generators::makeDoc(filter, rng);
+            bsoncxx::builder::stream::document document{};
+            doc->view(document);
+            _writeOps.push_back(mongocxx::model::delete_one{document.extract()});
+        } else if (writeCommand == "deleteMany") {
+            auto filter = writeOp["Filter"];
+            if (!filter) {
+                throw InvalidConfigurationException("deleteMany' expects a 'Filter' field.");
+            }
+            auto doc = value_generators::makeDoc(filter, rng);
+            bsoncxx::builder::stream::document document{};
+            doc->view(document);
+            _writeOps.push_back(mongocxx::model::delete_many{document.extract()});
+        } else if (writeCommand == "replaceOne") {
+            auto filter = writeOp["Filter"];
+            auto replacement = writeOp["Replacement"];
+            if (!filter || !replacement) {
+                throw InvalidConfigurationException(
+                    "'replaceOne' expects 'Filter' and 'Replacement' fields.");
+            }
+            auto filterTemplate = value_generators::makeDoc(filter, rng);
+            bsoncxx::builder::stream::document filterDocument{};
+            filterTemplate->view(filterDocument);
+            auto replacementTemplate = value_generators::makeDoc(replacement, rng);
+            bsoncxx::builder::stream::document replacementDocument{};
+            replacementTemplate->view(replacementDocument);
+            _writeOps.push_back(mongocxx::model::replace_one(filterDocument.extract(),
+                                                             replacementDocument.extract()));
+        } else if (writeCommand == "updateOne") {
+            auto filter = writeOp["Filter"];
+            auto update = writeOp["Update"];
+            if (!filter || !update) {
+                throw InvalidConfigurationException(
+                    "'updateOne' expects 'Filter' and 'Update' fields.");
+            }
+            auto filterTemplate = value_generators::makeDoc(filter, rng);
+            bsoncxx::builder::stream::document filterDocument{};
+            filterTemplate->view(filterDocument);
+            auto updateTemplate = value_generators::makeDoc(update, rng);
+            bsoncxx::builder::stream::document updateDocument{};
+            updateTemplate->view(updateDocument);
+            _writeOps.push_back(
+                mongocxx::model::update_one(filterDocument.extract(), updateDocument.extract()));
+        } else if (writeCommand == "updateMany") {
```

Comment:
hrm the number of `else if`s here is kinda worrisome - does it make sense to use the same `map<string,callback>` pattern here as you do for the operation name?